### PR TITLE
Use CRA (Redo): Pt 2/3: Working Unit + Integration Tests

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR fixes Unit and Integration tests for CRA.**
- We extend from CRA's jest config
- CRA's jest config clears mocks between tests, this is why we add the `beforeEach` to the `notistack` mocks.

## How has this been tested?

- Automated tests that cover added/changed functionality

### CODE REVIEW

File: src/components/experiment-creation/BasicInfo.test.tsx [hunk 1]
```diff
@@ -45,6 +45,8 @@ test('renders sensible dates as expected', () => {
 
   act(() => {
     fireEvent.change(startDateInput, { target: { value: '2020-07-28' } })
+  })
+  act(() => {
```
Comment (jessie-ross) [inline src/components/experiment-creation/BasicInfo.test.tsx:49]: Ah I accidentally left this in while trying to fix a warning - the only new warning overall, this didn't end up fixing it.

File: src/components/ConclusionsPanel.test.tsx [hunk 2]
```diff
@@ -12,10 +12,12 @@ import ConclusionsPanel from './ConclusionsPanel'
 
 jest.mock('notistack')
 const mockedNotistack = notistack as jest.Mocked<typeof notistack>
-mockedNotistack.useSnackbar.mockImplementation(() => ({
-  enqueueSnackbar: jest.fn(),
-  closeSnackbar: jest.fn(),
-}))
+beforeEach(() => {
```
Comment (yanirs) [inline src/components/ConclusionsPanel.test.tsx:15]: Can we use this opportunity to move this mocking boilerplate to `src/test-helpers/unit-test-setup.ts`?

File: src/components/experiment-creation/BasicInfo.test.tsx (hunk 1 above)
Comment (yanirs) [reply src/components/experiment-creation/BasicInfo.test.tsx:49]: Let's revert it then :slightly_smiling_face:

**Review by yanirs — APPROVED** [review]
Looks good, but please see inline comments before merging.

File: src/components/ConclusionsPanel.test.tsx (hunk 2 above)
Comment (jessie-ross) [reply src/components/ConclusionsPanel.test.tsx:15]: Looks like this won't work as mocks now reset between tests.

Comment (jessie-ross) [reply src/components/ConclusionsPanel.test.tsx:15]: Scratch that, I have figured out how to work it

**Comment by jessie-ross** [comment]
Merged in #390

**Pull request closed without merging**